### PR TITLE
First pass - Type errors

### PR DESCRIPTION
**Pull request by ostrokach**

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -106,14 +106,14 @@ a value which is an integer, not a function.
 
 ### E1111: Assignment from no return {#E1111}
 
-This error occurs when you assign a variable to the return value of a function call,
-but the function never returns anything.
-This error is similar to E1128. In the following example,
-f() does not return anything. As a result, `x` always gets the value `None`.
+This error occurs when you assign a variable to the return value of a function call, but the function never returns anything. This error is similar to [E1128](#E1128).
+
+In the following example, `add_fruit` modifies `fruit_basket` inplace and does not return anything. As a result, `new_fruit_basket` always gets the value `None`.
```
Comment (david-yz-liu) [inline index.md]: Use the term "mutate" rather than "modify," and don't use the term "inplace."

File: index.md [hunk 2]
```diff
@@ -106,14 +106,14 @@ a value which is an integer, not a function.
...
 
 ~~~~ {include="E1111_assignment_from_no_return"}
 ~~~~
 
+You should either modify `add_fruit` to return a new list, or call `add_fruit` without assigning the return value to a variable.
```
Comment (david-yz-liu) [inline index.md]: Let's use the first-person voice for these examples: "We should either..."

**Review by david-yz-liu — CHANGES_REQUESTED** [review]
Great work. I only left a few minor comments.

**Review by david-yz-liu — APPROVED** [review]

File: index.md [hunk 3]
```diff
@@ -187,41 +187,36 @@ print(a[0:3])
...
 
 ~~~~ {include="E1131_unsupported_binary_operation"}
 ~~~~
 
 
 ### E1135: Unsupported membership test {#E1135}
 
-This error occurs when you use the membership test `a in b`, but the `b`'s type
-does not support this type of check.
+This error occurs when you use the membership test `a in b`, but the type of `b`
```
Comment (nigef) [inline index.md]: Please avoid using the word "you", here and in other instances.  I believe this has been mentioned before.
There are many ways a phrase could be reworded, for example, 
> This error occurs when the membership of `a` is tested in `b`, i.e.  `a in b`, but the type of `b` does not support membership tests.

File: index.md [hunk 4]
```diff
@@ -1343,27 +1338,27 @@ print 3   # Error on this line
...
-        
+
         ```python
         class = 'algebra' # Error on this line
         ```
-        
+
     e. There is no ++ increment or –- decrement operator. Do not try to increment
-    or decrement a variable with ++ or --.
+    or decrement a variable with ++ or --.~~~~
```
Comment (nigef) [inline index.md]: typo?

File: index.md [hunk 5]
```diff
@@ -106,14 +106,14 @@ a value which is an integer, not a function.
...
-This error is similar to E1128. In the following example,
-f() does not return anything. As a result, `x` always gets the value `None`.
+This error occurs when you assign a variable to the return value of a function call, but the function never returns anything. This error is similar to [E1128](#E1128).
+
+In the following example, `add_fruit` mutates `fruit_basket` and does not return anything. As a result, `new_fruit_basket` always gets the value `None`.
 
 ~~~~ {include="E1111_assignment_from_no_return"}
 ~~~~
 
+We should either modify `add_fruit` to return a new list, or call `add_fruit` without assigning the return value to a variable.
```
Comment (nigef) [inline index.md]: Please reword to not use "we".

**Pull request merged**